fenix/app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt

```diff
@@ -271,55 +275,73 @@ fun PocketStories(
     val listState = rememberLazyListState()
     val flingBehavior = EagerFlingBehavior(lazyRowState = listState)
 
-    LazyRow(
-        modifier = Modifier.semantics {
-            testTagsAsResourceId = true
-            testTag = "pocket.stories"
-        },
-        contentPadding = PaddingValues(horizontal = contentPadding),
-        state = listState,
-        flingBehavior = flingBehavior,
-        horizontalArrangement = Arrangement.spacedBy(8.dp),
-    ) {
-        itemsIndexed(storiesToShow) { columnIndex, columnItems ->
-            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
-                columnItems.forEachIndexed { rowIndex, story ->
-                    Box(
-                        modifier = Modifier.semantics {
-                            testTagsAsResourceId = true
-                            testTag = when (story) {
-                                placeholderStory -> "pocket.discover.more.story"
-                                is PocketRecommendedStory -> "pocket.recommended.story"
-                                else -> "pocket.sponsored.story"
-                            }
-                        },
-                    ) {
-                        if (story == placeholderStory) {
-                            ListItemTabLargePlaceholder(stringResource(R.string.pocket_stories_placeholder_text)) {
-                                onDiscoverMoreClicked("https://getpocket.com/explore?$POCKET_FEATURE_UTM_KEY_VALUE")
-                            }
-                        } else if (story is PocketRecommendedStory) {
-                            PocketStory(
-                                story = story,
-                                backgroundColor = backgroundColor,
-                            ) {
-                                val uri = Uri.parse(story.url)
-                                    .buildUpon()
-                                    .appendQueryParameter(URI_PARAM_UTM_KEY, POCKET_STORIES_UTM_VALUE)
-                                    .build().toString()
-                                onStoryClicked(it.copy(url = uri), rowIndex to columnIndex)
-                            }
-                        } else if (story is PocketSponsoredStory) {
-                            Box(
-                                modifier = Modifier.onShown(0.5f) {
-                                    onStoryShown(story, rowIndex to columnIndex)
-                                },
-                            ) {
-                                PocketSponsoredStory(
+    BoxWithConstraints {
+        // BoxWithConstraints is re-composed as expected with screen change events; however,
+        // LocalConfiguration.current is not, hence why we're utilising getWindowVisibleDisplayFrame instead.
+        val rect = Rect()
+        composeView?.getWindowVisibleDisplayFrame(rect)
+        val screenHeightDp = rect.bottom.dp
```

Comment:
Kinda confused by this. so is neither `LocalConfiguration` nor `BoxWithConstraints` working to get the screen dimensions after recompositions?

I'd like to avoid passing in Views into Composables, so if we end up needing to go this route, I'd probably prefer to pass in a lambda instead.

Reply:
`BoxWithConstraints` is getting 'its' and 'its constraints' dimensions correctly. `LocalConfiguration` is not returning the updated layout specs on orientation change. I'd be interested to pair if you have any suggestions. Some logging I've been using in `BoxWithConstraints` to check the values.

```
Log.log(
    tag = "1821724",
    message = "Constraints - max width:${constraints.maxWidth.dp}, min width:${constraints.minWidth.dp}, max height:${constraints.maxHeight.dp}, min height:${constraints.minHeight.dp}",
)
Log.log(
    tag = "1821724",
    message = "Box - max width:$maxWidth, min width:$minWidth, max height:$maxHeight, min height:$minHeight",
)
Log.log(
    tag = "1821724",
    message = "Local config - screen height:${LocalConfiguration.current.screenHeightDp}, screen width:${LocalConfiguration.current.screenWidthDp}, orientation:${LocalConfiguration.current.orientation}",
)
```

Reply:
cc @rahulsainani
Update: The LocalConfiguration.current doesn't update (e.g. on orientation change) when there is only one language in the device system languages on API 33. Initial hypothesis is that it is related to [app language preferences](https://developer.android.com/guide/topics/resources/app-languages#impl-overview) that were added in API 33 A spike bug will investigate this.

Reply:
Bug created [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1836991)